### ISSUE TITLE
Empty files don't seem to be supported

### ISSUE THREAD

**Issue by proper1g1r** [optyfr/JRomManager#252]
I'm using the latest version (v3.0.0-beta.16), and any time a DAT file references an empty file, JRomManager flags it as unneeded:

<img width="897" src="https://github.com/optyfr/JRomManager/assets/158083504/f2797994-0fbb-4193-a1ef-f7771ad6a3c3">

<img width="1026" src="https://github.com/optyfr/JRomManager/assets/158083504/e6b4dd5a-4393-4a8e-a03c-3a0b5ba65b69">

All of those files are listed in the DAT file:

```xml
<game name="Need For Speed - Carbon - Domine a Cidade (Brazil) (Es,Pt)" id="0020">
  <description>Need For Speed - Carbon - Domine a Cidade (Brazil) (Es,Pt)</description>
  <!-- other non-empty files -->
  <rom name="mod\nfsresources\fe_fronta.bun" size="0" crc="00000000" md5="d41d8cd98f00b204e9800998ecf8427e" sha1="da39a3ee5e6b4b0d3255bfef95601890afd80709" sha256="e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"/>
  <rom name="mod\nfsresources\global\globala.bun" size="0" crc="00000000" md5="d41d8cd98f00b204e9800998ecf8427e" sha1="da39a3ee5e6b4b0d3255bfef95601890afd80709" sha256="e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"/>
  <rom name="mod\nfsresources\global\ingamea.bun" size="0" crc="00000000" md5="d41d8cd98f00b204e9800998ecf8427e" sha1="da39a3ee5e6b4b0d3255bfef95601890afd80709" sha256="e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"/>
</game>

<game name="Pac-Mania (Brazil) (Es,Pt)" id="0021">
  <description>Pac-Mania (Brazil) (Es,Pt)</description>
  <!-- other non-empty files -->
  <rom name="mod\276212\pm.bin" size="0" crc="00000000" md5="d41d8cd98f00b204e9800998ecf8427e" sha1="da39a3ee5e6b4b0d3255bfef95601890afd80709" sha256="e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"/>
</game>
```

And everything (CRC, MD5, SHA256) matches (and they're always the same, as they're all just empty files).

You can recreate the real world example I'm showing here by getting a DAT file for Zeebo [here](https://datomatic.no-intro.org/index.php?page=download&op=dat&s=126) and downloading the ROMs [here](https://archive.org/details/ni-romsets). I've seen this happening for 4 different No-Intro systems:

- Nintendo - Nintendo DSi (Digital) (CDN) (Decrypted)
- Nintendo - Nintendo DSi (Digital) (CDN) (Encrypted)
- Tiger - Gizmondo
- Zeebo - Zeebo

**Comment by cinnamonmatexfce** [comment]
Try enabling *Calculate All SHA1*

**Comment by tfcporciuncula** [comment]
It doesn't seem to make a difference and I get the same result. If I enable _Ignore Unneeded Entries_ then the report becomes empty, but the _Profiles_ tab still shows it as incomplete (e.g. _HaveSets_ 59/61 and _HaveRoms_ 2755/2759 in this case).

**Comment by proper1g1r** [comment]
It doesn't seem to make a difference and I get the same result. If I enable _Ignore Unneeded Entries_ then the report becomes empty, but the _Profiles_ tab still shows it as incomplete (e.g. _HaveSets_ 59/61 and _HaveRoms_ 2755/2759 in this case).

**Comment by cinnamonmatexfce** [comment]
Can you give [latest stable](https://github.com/optyfr/JRomManager/releases/download/2.5.1/JRomManager-2.5.1-x86_64.AppImage) a try?
If yes, check the behaviour with SHA1 enabled and disabled also.

**Comment by proper1g1r** [comment]
I'm actually unable to install it. When I try to open the `.app` file from the `.dmg`, I get this message:

>“JRomManager.app” is damaged and can’t be opened. You should move it to the Bin.

And I'm not sure what to do with the `.AppImage` file you linked.

Edit: I'm on macOS Ventura 13.5 with an M2 Max, should've mentioned it from the start, sorry!

**Comment by optyfr** [comment]
Ok, forget the SHA1 thing, there is no reason it would change anything, and stable version neither

Zero sized files are intentionally considered as unneeded... because, for mame dat files, they are really unneeded (ok they should be flagged as nodump or badump but a lot of dats didn't care and simply set them to zero)... and here come the "ignore unneeded entries" purpose : to not delete those files, but they are still considered as unneeded anyway... 

Now, just to be sure, are those zero files currently there on your disk? There is currently no option to create them, but maybe an option to explicitely tell that they are finally needed should come there, just tell me if they are vital for your set

**Comment by proper1g1r** [comment]
Yes, the empty files are in my disk, but I actually have no idea if they're vital for the game or not, and there's no easy way for me to test that right now. However, they are referenced in DAT files from No-Intro, so I would assume they're relevant.

I think maybe it would be nice to have a separate "Ignore Empty Files" options, so you could still ditch actual unneeded files but keep empty files referenced in DAT files around. Right now if a DAT entry references empty files, there's no way to make JRomManager happy since first it'll complain the empty files are unneeded, and then if you "fix" it, it'll delete the files but then complain they're missing (but not in the report, which will remain blank, just in the profile tab or in the profile viewer).

**Comment by optyfr** [comment]
In that case, it would be more than "ignore Empty Files", because if we assume they are needed for a certain type of dat or emulator, then we have to create them if they are missing, so it would be something like  "Assume Empty Files as Needed", btw it's a special case because even if we don't have empty files in source, they can perfectly be created anyway (and probably also some known files filled with zero bytes too but that's another story).

**Comment by tfcporciuncula** [comment]
Yup, that makes sense! And I would even consider having this on by default to avoid this "can't get JRomManager happy if DAT references empty files" scenario. In most cases DAT files won't reference empty files and they can be removed by JRomManager anyway.

**Comment by proper1g1r** [comment]
Yup, that makes sense! And I would even consider having this on by default to avoid this "can't get JRomManager happy if DAT references empty files" scenario. In most cases DAT files won't reference empty files and they can be removed by JRomManager anyway.

**Comment by optyfr** [comment]
resolved in commit bd7f8f3
